### PR TITLE
Add sourcelink support

### PR DESCRIPTION
**Pull request by KristianJakubik**
## Description

I've added support for SourceLink. Now consumer of this package should be able to easily debug through the cosmos sdk.

I've tested it only manually. I tried to be as close as possible to azure pipeline tasks by running `dotnet cli` locally with arguments from tasks but I will still suggest to test it once again with normal azure pipelines and see if it is working correctly.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Closing issues

#1069

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <PackageReleaseNotes>The change log for this SDK is made available at https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/changelog.md at the time of release.</PackageReleaseNotes>
     <PackageLicenseUrl>https://aka.ms/netcoregaeula</PackageLicenseUrl>
     <PackageProjectUrl>https://github.com/Azure/azure-cosmos-dotnet-v3</PackageProjectUrl>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj]: Can you please add a note in the changelog.

Comment (KristianJakubik) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj]: Sure. Done 🙂.

**Pull request merged**